src/main/java/mate/academy/util/HashUtil.java

```diff
@@ -0,0 +1,32 @@
+package mate.academy.util;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+
+public class HashUtil {
+    private static final SecureRandom RANDOM = new SecureRandom();
+    private static final String HASH_ALGORITHM = "SHA-256";
+
+    public static String hashPassword(String password) {
+        byte[] salt = new byte[16];
+        RANDOM.nextBytes(salt);
+        return hash(password, salt);
+    }
+
+    public static boolean isValidPassword(String password, String hashedPassword) {
+        String salt = hashedPassword.substring(0, 16);
+        return hashedPassword.equals(hash(password, salt.getBytes()));
```

Comment:
The `isValidPassword` method incorrectly assumes that the salt is the first 16 characters of the hashed password. Ensure that the salt is stored separately or encoded in a way that can be reliably extracted for validation.